tests/core/test_emitter.py

```diff
@@ -17,3 +18,23 @@ def test_remove_control_chars(self):
 
         for bad, good in messages:
             self.assertTrue(remove_control_chars(bad) == good, (bad,good))
+  
+    def test_remove_control_chars_from(self):
+        bad_messages = [
+            { "processes":[1234,[["d≤Ω≈ç√∫˜µ≤", 0, 2.2,12,34,"compiz",1]]]}
+            ]
+        good_messages = [
+            { "processes":[1234,[["db", 0, 2.2,12,34,"compiz",1]]]}
+            ]
+
+        def is_converted_same(msg):
+            new_msg = remove_control_chars_from(msg, None)
+            if str(new_msg) == str(msg):
+                return True
+            return False
+
+        for msg in bad_messages:
+            self.assertFalse(is_converted_same(msg))
```

Comment:
you should also test for an expected result i.e. in this case, the function could return something completely wrong and this test would still pass.

You should also add a few more cases but otherwise 👍